### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by MichalBryxi** [ember-template-lint/ember-template-recast#586]
Somewhere between `v5.0.1` and `v5.0.3` we got an issue that manifests as follows (log taken from `ember-template-lint-plugin-tailwindcss`):

```
    +     "message": "Maximum call stack size exceeded",
    +     "severity": 2,
    +     "source": "RangeError: Maximum call stack size exceeded
    +     at Object.get [as print] (/Users/michal/ember/ember-template-lint-plugin-tailwindcss/node_modules/ember-template-recast/node_modules/@glimmer/syntax/dist/commonjs/es2017/index.js:78:17)
    +     at print (/Users/michal/ember/ember-template-lint-plugin-tailwindcss/node_modules/ember-template-recast/src/index.ts:21:12)
    +     at Object.override (/Users/michal/ember/ember-template-lint-plugin-tailwindcss/node_modules/ember-template-recast/src/index.ts:25:18)
    +     at Printer.print (/Users/michal/ember/ember-template-lint-plugin-tailwindcss/node_modules/ember-template-recast/node_modules/@glimmer/syntax/packages/@glimmer/syntax/lib/generation/printer.ts:539:20)
    +     at Object.build (/Users/michal/ember/ember-template-lint-plugin-tailwindcss/node_modules/ember-template-recast/node_modules/@glimmer/syntax/packages/@glimmer/syntax/lib/generation/print.ts:13:10)
    +     at print (/Users/michal/ember/ember-template-lint-plugin-tailwindcss/node_modules/ember-template-recast/src/index.ts:21:12)
    +     at Object.override (/Users/michal/ember/ember-template-lint-plugin-tailwindcss/node_modules/ember-template-recast/src/index.ts:25:18)
    +     at Printer.print (/Users/michal/ember/ember-template-lint-plugin-tailwindcss/node_modules/ember-template-recast/node_modules/@glimmer/syntax/packages/@glimmer/syntax/lib/generation/printer.ts:539:20)
```

**Comment by rwjblue** [comment]
Do you happen to know the steps to repro? Our internal tests don't seem to hit this, it's going to be hard to figure out what is wrong without some high level understanding of what manipulation causes it... 🤔 

Maybe you could copy a test like this one, to replicate?

https://github.com/ember-template-lint/ember-template-recast/blob/ec7c070c73db48a122dcf6608e5653de10226fa4/src/parse-result.test.ts#L16-L26

**Comment by MichalBryxi** [comment]
No idea yet. It might even be a complete red herring as only when I [shake the yarn.lock](https://gitlab.com/michal-bryxi/open-source/ember-template-lint-plugin-tailwindcss/-/commit/96dcce4557dd809b5c8c79ab1874d26347607a68) I get the problematic behaviour. I will try to get a failing test for you.

**Comment by MichalBryxi** [comment]
I could trace it to [this recursive print() function](https://github.com/ember-template-lint/ember-template-recast/blob/master/src/index.ts#L25). When I try to `console.log(ast)` I get something like:

```js
{
        type: 'ConcatStatement',
        parts: [
          {
            type: 'TextNode',
            chars: '\n' +
              '      bazDoo\n' +
              '      gap-4 grid grid-cols-2\n' +
              '      lg:grid-cols-5 md:grid-cols-3 sm:grid-cols-3 xl:grid-cols-6\n' +
              '      ',
            loc: [Object]
          },
          {
            type: 'MustacheStatement',
            path: [Object],
            params: [Array],
            hash: [Object],
            escaped: true,
            loc: [Object],
            strip: [Object]
          },
          { type: 'TextNode', chars: '\n      ', loc: [Object] },
          {
            type: 'MustacheStatement',
            path: [Object],
            params: [Array],
            hash: [Object],
            escaped: true,
            loc: [Object],
            strip: [Object]
          },
          { type: 'TextNode', chars: '\n    ', loc: [Object] }
        ],
        loc: {
          source: null,
          start: { line: 3, column: 10 },
          end: { line: 13, column: 5 }
        }
}
```

Which seems reasonable to me. But I don't understand why it's recursing infinitely nor what's the logic of that function in the first place.

**Comment by MichalBryxi** [comment]
I think I found a repro: https://github.com/ember-template-lint/ember-template-recast/pull/589

Seems like when I create new `concat` with `parts` from previous `concat` bad things will happen. cc: @rwjblue

**Comment by MichalBryxi** [comment]
Sounds like this one was resolved with #671
